### PR TITLE
Add article on how to a create key pair for a workload cluster via the Management API

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/17727 , https://github.com/giantswarm/giantswarm/issues/17899

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [x] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+last_review_date: 2021-07-26
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-biscuit
+---
+
+# Creating workload cluster key pairs via the Management API
+
+We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a method specific to Giant Swarm to create such key pairs via the Management API.
+
+**Note:** The method described here makes use of the [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) <abbr title="custom resource definition">CRD</abbr></a> and [cert-operator](https://github.com/giantswarm/cert-operator), which are part of every Giant Swarm management cluster currently. As Giant Swarm is shifting towards using Cluster API controllers, this method is not guaranteed to work once this transition is done. Please consider it a temporary solution.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:21]:
```suggestion
**Note:** The method described here makes use of the [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) <abbr title="custom resource definition">CRD</abbr></a> and [cert-operator](https://github.com/giantswarm/cert-operator), which are currently part of every Giant Swarm management cluster. As Giant Swarm is shifting towards using Cluster API controllers, this method will not work once this transition is done. Please consider it a temporary solution.
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+**Note:** If you want to use the deprecated Giant Swarm Rest API for creating key pairs, please check [`gsctl create kubeconfig`]({{< relref "/ui-api/gsctl/create-kubeconfig.md" >}}) and [`gsctl create keypair`]({{< relref "/ui-api/gsctl/create-keypair.md" >}}).
+
+## Prerequisites
+
+You need:
+
+- Your installation's Management API endpoint (similar to `https://g8s.codename.eu-west-1.aws.gigantic.io`)or base domain (similar to `codename.eu-west-1.aws.gigantic.io`).
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:29]:
```suggestion
- Your installation's Management API endpoint (similar to `https://g8s.codename.eu-west-1.aws.gigantic.io`) or base domain (similar to `codename.eu-west-1.aws.gigantic.io`).
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+Using our `kubectl` plug-in, you can make sure to be authenticated against the Management API:
+
+```nohighlight
+kubectl gs login <URL>
+```
+
+Here, `<URL>` can be either
+
+- the Management API endpoint URL or host name
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:54]: what does 'URL or host name' mean here?

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+- your installation's web interface URL or host name
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:55]: do we explicitly refer to happa elsewhere, in a situation like this?

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+Here is an example of a `CertConfig` resource manifest. Follow the detailed explanations below.
+
+```yaml
+apiVersion: core.giantswarm.io/v1alpha1
+kind: CertConfig
+metadata:
+  name: mycert-20210726
+  namespace: default
+  labels:
+    cert-operator.giantswarm.io/version: 0.1.0
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:72]: It's pretty brutal customers have to care about this :D

File: src/content/ui-api/management-api/wc-key-pairs/index.md (hunk 3 above)
Comment (marians) [reply src/content/ui-api/management-api/wc-key-pairs/index.md:54]: It means you can use either `https://blah` or `blah`

File: src/content/ui-api/management-api/wc-key-pairs/index.md (hunk 4 above)
Comment (marians) [reply src/content/ui-api/management-api/wc-key-pairs/index.md:55]: No, we call it the web interface officially and even decided to not use the term "happa" any more in communication.

File: src/content/ui-api/management-api/wc-key-pairs/index.md (hunk 5 above)
Comment (marians) [reply src/content/ui-api/management-api/wc-key-pairs/index.md:72]: Does this need explanation? Do we have to update this every time cert-operator is released? Do we want to remove this and default it via an admission controller?

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
+---
+linkTitle: Key pairs for workload clusters
+title: Creating workload cluster key pairs via the Management API
+description: We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a Giant Swarms specific way to create such key pairs via the Management API.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:4]:
```suggestion
description: We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a Giant Swarm specific way to create such key pairs via the Management API.
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md (hunk 4 above)
Comment (JosephSalisbury) [reply src/content/ui-api/management-api/wc-key-pairs/index.md:55]: Works for me

**Comment by marians** [comment]
@JosephSalisbury So would you approve, or is there more to do?

File: src/content/ui-api/management-api/wc-key-pairs/index.md (hunk 5 above)
Comment (JosephSalisbury) [reply src/content/ui-api/management-api/wc-key-pairs/index.md:72]: > Does this need explanation?

i have a hunch we can get away with it

> Do we have to update this every time cert-operator is released?

We'd need to update it if we get rid of this cert-operator version, which i don't see happening until we just get rid of cert-operator entirely.

> Do we want to remove this and default it via an admission controller?

Not really :D

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+  versionBundle:
+    version: 0.1.0
+```
+
+Here is how you should adapt the example to match your case:
+
+- `.metadata`:
+    - `.name`: name for this `CertConfig` resource, unique within the management cluster.
+    - `.labels`:
+        - `giantswarm.io/certificate`: an identifier for this key pair that must be unique within the management cluster.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:96]:
```suggestion
        - `.giantswarm.io/certificate`: an identifier for this key pair that must be unique within the management cluster.
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        - `giantswarm.io/cluster`: ID of the workload cluster to grant access to.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:97]:
```suggestion
        - `.giantswarm.io/cluster`: ID of the workload cluster to grant access to.
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+        - `giantswarm.io/organization`: Name of the organization owning the above workload cluster.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:98]:
```suggestion
        - `.giantswarm.io/organization`: Name of the organization owning the above workload cluster.
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+- `.spec`:
+    - `.cert`:
+        - `.clusterComponent`: set this to the same value as the `giantswarm.io/certificate` label.
+        - `.clusterID`: set this to the same value as the `giantswarm.io/cluster` label.
+        - `.commonName`: the `CN` that the created certificate will have. Set this to a unique string ending with your workload cluster's Kubernetes API endpoint domain.
+        - `.organizations`: defines the RBAC groups set in the created certificate via the `O` attribute. You can include `system:masters` here to create an admin key pair. Check the security notice below.
+        - `ttl`: The duration until the created certificate will expire. `1h` stands for one hour. Check the security notice below.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:105]:
```suggestion
        - `.ttl`: The duration until the created certificate will expire. `1h` stands for one hour. Check the security notice below.
```

File: src/content/ui-api/management-api/wc-key-pairs/index.md [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+last_review_date: 2021-07-26
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-biscuit
+---
+
+# Creating workload cluster key pairs via the Management API
+
+We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a method specific to Giant Swarm to create such key pairs via the Management API.
+
+**Note:** The method described here makes use of the [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) <abbr title="custom resource definition">CRD</abbr></a> and [cert-operator](https://github.com/giantswarm/cert-operator), which are currently part of every Giant Swarm management cluster. As Giant Swarm is shifting towards using Cluster API controllers, this method will not work once this transition is done. Please consider it a temporary solution.
```
Comment (JosephSalisbury) [inline src/content/ui-api/management-api/wc-key-pairs/index.md:21]:
```suggestion
**Note:** The method described here makes use of the [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) <abbr title="custom resource definition">CRD</abbr></a> and [cert-operator](https://github.com/giantswarm/cert-operator), which is currently part of every Giant Swarm management cluster. As Giant Swarm is shifting towards using Cluster API controllers, this method will not work once this transition is done. Please consider it a temporary solution.
```

**Review by JosephSalisbury — APPROVED** [review]
lets goo

**Pull request merged**